jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/CreateEcrRepoDialog.kt

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.ecr
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import org.jetbrains.annotations.TestOnly
+import software.amazon.awssdk.services.ecr.EcrClient
+import software.aws.toolkits.jetbrains.core.explorer.refreshAwsTree
+import software.aws.toolkits.jetbrains.services.ecr.resources.EcrResources
+import software.aws.toolkits.jetbrains.utils.ui.validationInfo
+import software.aws.toolkits.resources.message
+import software.aws.toolkits.telemetry.EcrTelemetry
+import software.aws.toolkits.telemetry.Result
+import java.awt.Component
+import javax.swing.JComponent
+
+class CreateEcrRepoDialog(
+    private val project: Project,
+    private val ecrClient: EcrClient,
+    parent: Component? = null,
+    initialRepoPolicy: String = ""
+) : DialogWrapper(project, parent, false, IdeModalityType.PROJECT) {
+    private val view = CreateRepoPanel(project, initialRepoPolicy)
+
+    var repoName
+        get() = view.repoName.text.trim()
```

Comment:
i'm assuming spaces aren't allowed

Reply:
yep, spaces are not allowed, here's the disgusting regex the service uses:
```
/^(?:[a-z0-9]+(?:[._-][a-z0-9]+)*\/)*[a-z0-9]+(?:[._-][a-z0-9]+)*$/
```